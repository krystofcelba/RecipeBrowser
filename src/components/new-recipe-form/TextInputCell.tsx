import React from 'react';
import { Cell } from 'react-native-tableview-simple';
import { Field } from 'redux-form/immutable';

import { View, BoldText } from 'src/components/common';
import AppTheme from 'src/assets/appTheme';

import FormTextInput from './FormTextInput';

interface Props {
  index?: number;
  name: string;
  placeholder?: string;
  numberOfLines?: number;
}

const TextInputCell = ({ index, ...otherProps }: Props) => {
  const showIndex = index !== undefined;
  return (
    <Cell
      cellContentView={
        <View flexDirection="row" flex={1} paddingVertical={AppTheme.defaultPadding / 2}>
          {showIndex && <BoldText>#{index + 1}</BoldText>}
          <Field
            component={FormTextInput}
            {...otherProps}
            style={showIndex && { paddingLeft: AppTheme.defaultPadding }}
          />
        </View>
      }
    />
  );
};

export default TextInputCell;
