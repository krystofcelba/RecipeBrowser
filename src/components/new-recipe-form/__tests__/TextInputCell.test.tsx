import React from 'react';
import { shallow } from 'enzyme';

import TextInputCell from '../TextInputCell';

describe('FormInputCell component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<TextInputCell name="input" placeholder="placeholder" value="" onChangeText={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
