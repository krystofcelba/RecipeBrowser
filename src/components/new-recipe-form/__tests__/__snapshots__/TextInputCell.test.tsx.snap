// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputCell component renders as expected 1`] = `
<Cell
  accessory={false}
  accessoryColor="#007AFF"
  allowFontScaling={true}
  backgroundColor="#FFF"
  cellAccessoryView={null}
  cellContentView={
    <glamorous.View
      flex={1}
      flexDirection="row"
      paddingVertical={5}
    >
      <Field
        component={[Function]}
        name="input"
        onChangeText={[Function]}
        placeholder="placeholder"
        style={false}
        value=""
      />
    </glamorous.View>
  }
  cellImageView={null}
  cellStyle="Basic"
  contentContainerStyle={Object {}}
  detail=""
  detailTextStyle={Object {}}
  disableImageResize={false}
  highlightActiveOpacity={0.8}
  highlightUnderlayColor="black"
  image={null}
  isDisabled={false}
  leftDetailColor="#007AFF"
  onHighlightRow={null}
  onPress={false}
  onUnHighlightRow={null}
  rightDetailColor="#8E8E93"
  subtitleColor="#000"
  subtitleTextStyle={Object {}}
  title=""
  titleTextColor="#000"
  titleTextStyle={Object {}}
  titleTextStyleDisabled={Object {}}
/>
`;
